controllers/control_plane_init_locker.go

```diff
@@ -35,16 +36,18 @@ type ControlPlaneInitLocker interface {
 
 // controlPlaneInitLocker uses a ConfigMap to synchronize cluster initialization.
 type controlPlaneInitLocker struct {
-	log             logr.Logger
-	configMapClient corev1.ConfigMapsGetter
+	log    logr.Logger
+	client client.Client
+	ctx    context.Context
 }
 
 var _ ControlPlaneInitLocker = &controlPlaneInitLocker{}
 
-func newControlPlaneInitLocker(log logr.Logger, configMapClient corev1.ConfigMapsGetter) *controlPlaneInitLocker {
+func newControlPlaneInitLocker(ctx context.Context, log logr.Logger, client client.Client) *controlPlaneInitLocker {
```

Comment:
We should do this afterwards, but I think this code should move into its own package under `controllers/`, seems a little out of place.

Reply:
I will move the init locker into a package under `controllers/` in a different PR.